### PR TITLE
chore: imporved deployement script

### LINKED ISSUE #217: Improve `deploy` function on `deploy-contract.ts`
## Issue Overview

- update the order of functions, able to receive null constructor params
- the only `required` param is the contractName.
- receives the contract name first, then optional `exportContractName`, then the optional`constructor params`,  and finally the `options`
- makes good usage of  `options` and `maxFee`
- remove unused imports
- good error logging for all cases !
- other dev improvements ++

## Proposed Solutions or Ideas

- should be tested on all networks.

### PR DESCRIPTION
**Pull request by ikemHood**
# Improve deploy function on deploy-contract.ts

Fixes #217 

utilized maxFee
removed unused imports
imporved logging
tested on devnet and sepolia

- [ ] Feature
- [ ] Bug
- [✅ ] Enhancement

### CODE REVIEW

File: packages/snfoundry/scripts-ts/deploy.ts
Comment (0xquantum3labs) [inline packages/snfoundry/scripts-ts/deploy.ts:1]: i should be able to call this 
```javascript
  await deployContract(
    "YourContract",
    "YourContractExportName",
    {
      owner: deployer.address
    },
    {
      maxFee: BigInt(1000000000000)
    }
  );
```

also this  ( THIS IS NOT POSSIBLE YET , IT ENFORCES A EXPORT NAME )

```javascript
  await deployContract(
    "YourContract",
    {
      owner: deployer.address,
    }
```
and also this for a contract that doesnt receive a constructor

```javascript
  await deployContract(
    "YourContract",
  );
```

Comment (0xquantum3labs) [inline packages/snfoundry/scripts-ts/deploy.ts:1]: lets add some instructions as comments on how to use this function with example within this `deployScript()` function,

File: packages/snfoundry/scripts-ts/deploy-contract.ts [hunk 1]
```diff
@@ -3,40 +3,37 @@ import path from "path";
...
 const networkName: string = argv["network"];
 
 let deployments = {};
-
 let deployCalls = [];
 
 const { provider, deployer }: Network = networks[networkName];
 
-const declareIfNot_NotWait = async (payload: any) => {
+const declareIfNot_NotWait = async (payload: DeclareContractPayload, options?: { maxFee?: bigint}) => {
```
Comment (0xquantum3labs) [inline packages/snfoundry/scripts-ts/deploy-contract.ts:23]: this is a `maxDeclareFee` rather than a `maxFee` at the moment.

Comment (0xquantum3labs) [reply packages/snfoundry/scripts-ts/deploy-contract.ts:23]: we can add another optional `maxFee` [here](https://github.com/ikemHood/scaffold-stark-2/blob/3d3aa29d02512650c459b8c38cfab4181be4bb1a/packages/snfoundry/scripts-ts/deploy-contract.ts#L178-L179) 

```javascript
const executeDeployCalls = async () => {

```

and this maxFee will be set up once inside `deploy.ts` and passed to this function when its called, should be nullable item and add instructions on `deploy.ts`

**Review by 0xquantum3labs — CHANGES_REQUESTED** [review]

File: packages/snfoundry/scripts-ts/helpers/parse-deployments.ts [hunk 2]
```diff
@@ -76,7 +76,7 @@ const generateTsAbis = () => {
 
   fs.writeFileSync(
     path.join(TARGET_DIR, "deployedContracts.ts"),
-    prettier.format(
+    await prettier.format(
```
[the remark this answers is not in the document]
Comment (ikemHood) [reply packages/snfoundry/scripts-ts/helpers/parse-deployments.ts:79]: removing that produces a type error
`Argument of type 'Promise<string>' is not assignable to parameter of type 'string | ArrayBufferView'.
  Type 'Promise<string>' is missing the following properties from type 'DataView': buffer, byteLength, byteOffset, getFloat32, and 19 more.`

File: packages/snfoundry/scripts-ts/deploy.ts
Comment (ikemHood) [reply packages/snfoundry/scripts-ts/deploy.ts:1]: This works?
```
  await deployContract(
    {
      contractName: "YourContract",
      exportContractName: "YourContractExportName",
      constructorArgs: {
        owner: deployer.address,
      },
      options:{
        maxFee: BigInt(1000000000000)
      }
    }
  );
```

File: packages/snfoundry/scripts-ts/deploy-contract.ts (hunk 1 above)
Comment (ikemHood) [reply packages/snfoundry/scripts-ts/deploy-contract.ts]: `executeDeployCalls ` usage doesn't make it straight forward to add this params, but i have added it regardless.

another option can be to set the options from deployContract to a variable just like the deployCalls and access it inside the function. Let me know if i should implement this

**Comment by 0xquantum3labs** [comment]
this PR shoudnt change anything on `packages/nextjs` please check the changed files.

File: .gitignore [hunk 3]
```diff
@@ -12,4 +12,7 @@ package-lock.json
 !.yarn/sdks
 !.yarn/versions
 .eslintcache
-.DS_Store
\ No newline at end of file
+.DS_Store
+
+# bun
+bun.lockb
```
Comment (0xquantum3labs) [inline .gitignore:18]: why do we have this ? @ikemHood

**Review by 0xquantum3labs — CHANGES_REQUESTED** [review]

**Comment by 0xquantum3labs** [comment]
this PR shoudnt change anything on packages/nextjs please check the changed files.
i still see changes on `packages/nextjs`

**Comment by 0xquantum3labs** [comment]
also please check failing CI

**Comment by Nadai2010** [comment]
Hi @ikemHood , you must add the removed files, even if they are auto-generated, you cannot remove anything from packages/nextjs, keep it the same as the main, it has already been left in the comments.

Please also review the rest of the comments, the tests continue to fail, once this is corrected, upload it with the correct CI

**Comment by 0xquantum3labs** [comment]
closing this as changes are still visible on  `packages/nextjs`, without updates .

**Pull request closed without merging**